test/test_sparse.py

```diff
@@ -3561,6 +3541,69 @@ def test(sparse_dims, nnz, with_size, new_size):
         test(4, 6, [7, 3, 1, 3, 1, 3], [7, 3, 1, 3, 2, 3])
         test(4, 6, [7, 3, 1, 3, 2, 1], [7, 3, 1, 3, 2, 3])
 
+    @coalescedonoff
+    @dtypes(*all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16))
+    @precisionOverride({torch.bfloat16: 1e-2, torch.float16: 1e-2})
+    def test_sparse_dense_mul(self, device, dtype, coalesced):
+        shape = (2, 3, 4, 10)
+        nnz = 10
+
+        def check(self, s, d):
+            res = d * s
+
+            # check commutativity
+            self.assertEqual(res, d * s)
+
+            # check correctness
+            self.assertEqual(res.to_dense(), s.to_dense() * d)
+
+            # check in-placeness for dense
+            if d.dim() >= s.dim():
+                dc = d.clone()
+                self.assertEqual(d.mul_(s), dc.mul_(s.to_dense()))
+
+            # check in-placeness for sparse
+            if s.dim() >= d.dim():
+                # for sparse
+                sc = s.clone()
+                self.assertEqual(s.mul_(d).to_dense(), sc.to_dense().mul_(d))
+
+        for dim in range(len(shape) + 1):
+            sub_shape = shape[dim:]
+            sparse_dim = len(sub_shape) // 2
+
+            def check_empty(sparse_shape, nnz, dense_shape, coalesce):
+                from itertools import product
+                for nnz_val, shape_suffix in product((nnz, 0), ((), (0,))):
+                    empty_sparse_shape = sparse_shape + shape_suffix
+                    empty_dense_shape = dense_shape + shape_suffix
+                    s = self._gen_sparse(sparse_dim, nnz_val, empty_sparse_shape, dtype, device, coalesce)[0]
+                    d = make_tensor(empty_dense_shape, dtype=dtype, device=device)
+                    check(self, s, d)
+
+
+            coalesce_sparse = coalesced
+            # This case always coalesce inputs and that could lead to loss of precision,
+            # hence it is inhibited for float16/bfloat16 by providing already coalesced tensors.
+            if dtype in {torch.float16, torch.bfloat16}:
+                coalesce_sparse = True
+            # to_dense is problematic for boolean non-coalesced CUDA tensors
+            # see https://github.com/pytorch/pytorch/issues/81648
+            if dtype == torch.bool and torch.device(device).type == "cuda":
+                coalesce_sparse = True
```

Comment:
General note: If `coalescedonoff` were to generate parameterized subtests like other decorators do you could simply `raise unittest.SkipTest(...)` here. Which would be preferable as the problematic dtype/device combinations would not need to run twice. 

As it `coalescedonoff` is currently implemented that would skip both `coalesced=True` and `coalesced=False` as they run both cases within the same test. 

You could still stop early here so that the dtype/coalesced/device combination does not execute twice. It could be confusing, but the `coalesce_sparse` value will be closer to the failure point so probably not.

Reply:
I agree in general, but in this case we still want bool tested with mul, and the issue comes from `to_dense`.